tedana/workflows/tedana.py

```diff
@@ -387,56 +405,75 @@ def tedana_workflow(data, tes, mask=None, mixm=None, ctab=None, manacc=None,
     if ctab and not mixm:
         LGR.warning('Argument "ctab" requires argument "mixm".')
         ctab = None
-    elif ctab and (manacc is None):
-        LGR.warning('Argument "ctab" requires argument "manacc".')
-        ctab = None
     elif manacc is not None and not mixm:
         LGR.warning('Argument "manacc" requires argument "mixm".')
         manacc = None
 
+    if t2smap is not None and op.isfile(t2smap):
+        t2smap = op.abspath(t2smap)
+        # Allow users to re-run on same folder
+        if t2smap != op.join(out_dir, 't2sv.nii'):
+            shutil.copyfile(t2smap, op.join(out_dir, 't2sv.nii'))
+            shutil.copyfile(t2smap, op.join(out_dir, op.basename(t2smap)))
+    elif t2smap is not None:
+        raise IOError('Argument "t2smap" must be an existing file.')
+
     bp_str = ("TE-dependence analysis was performed on input data.")
-    if mask is None:
+    if mask and not t2smap:
+        # TODO: add affine check
+        LGR.info('Using user-defined mask')
+        bp_str += (" A user-defined mask was applied to the data.")
+    elif t2smap and not mask:
+        LGR.info('Using user-defined T2* map to generate mask')
+        t2s = utils.load_image(t2smap)
+        t2sG = t2s.copy()
+        mask = (t2s != 0).astype(int)
+    elif t2smap and mask:
+        LGR.info('Using user-defined mask and T2* map to generate mask')
```

Comment:
Can we rephrase this ? It seems contradictory at first read-through.

Reply:
What about "*Combining* user-defined mask and T2* map to generate mask"?